mypyc/irbuild/ll_builder.py

```diff
@@ -623,12 +623,12 @@ def unary_op(self,
                  lreg: Value,
                  expr_op: str,
                  line: int) -> Value:
-        call_c_ops_candidates = c_unary_ops.get(expr_op, [])
-        target = self.matching_call_c(call_c_ops_candidates, [lreg], line)
-        if target:
-            return target
         ops = unary_ops.get(expr_op, [])
         target = self.matching_primitive_op(ops, [lreg], line)
+        if target:
+            return target
+        call_c_ops_candidates = c_unary_ops.get(expr_op, [])
+        target = self.matching_call_c(call_c_ops_candidates, [lreg], line)
```

Comment:
this adjustment is to ensure the only left old-style unary op(boolean `not`) is always used first then the generic one. 

The boolean `not` should eventually be built using a `ComparsionOp`(or maybe `UnaryComparisonOp` so we differentiate the unary/binary property?) Then this hack can be removed.

Reply:
Unary comparison op doesn't seem worth it.

Reply:
> Unary comparison op doesn't seem worth it.

Then probably a single `ComparisonOp` class for all ops producing boolean, with arg_types and op_type to ensure each operator gets the exact number of argument it needs?

Reply:
I meant that having binary comparison operations seems sufficient to me. The unary operations can be built using the binary operations.

Reply:
I see, a `NOR` operation should be sufficient to represent the `not` here.